### ISSUE TITLE
Mean and std for a different dataset

### ISSUE THREAD

**Issue by afshindn** [facebook/fb.resnet.torch#78]
Hi, 

How can I get the mean and std used in extract_features.lua for a different dataset? Are those numbers being saved somewhere during training?

Thanks

**Comment by colesbury** [comment]
I think you should use the mean and std used during training and not recompute it for your own dataset. (Although you can try that to see if it performs better). If you're using the pretrained models, the values in extract_features are accurate:

https://github.com/facebook/fb.resnet.torch/blob/8549d8b98ade293a07ac94b1494e344a22e6e27d/pretrained/extract-features.lua#L85-L88

matches the values used in training:

https://github.com/facebook/fb.resnet.torch/blob/8549d8b98ade293a07ac94b1494e344a22e6e27d/datasets/imagenet.lua#L67-L70

**Comment by afshindn** [comment]
Thanks Sam! 

I trained the model on my own dataset but the default 'dataset' option was set to "imagenet". So it used all the preprocessing you used for imagenet. Beside the mean which is a standard thing, could you gimme a bit more info about the other pre-processing options and how to compute them on my own data. Is there also a reason why you don't do the lighting as a preprocessing during feature extraction?

https://github.com/facebook/fb.resnet.torch/blob/8549d8b98ade293a07ac94b1494e344a22e6e27d/datasets/imagenet.lua#L67-L70

The other question I have is, I trained the 101 layer network on my own data, and each model snapshot is ~11GB. Any idea why it is so big compare to the released pertained models on imagenet?

Thanks

**Comment by colesbury** [comment]
The mean/std processing (`t.ColorNormalize`) is just for input normalization to improve training.

The other processing is intended for reducing overfitting by adding noise:

1. RandomSizedCrop(224): distorts aspect ratio and takes a random crop from the distorted image
2. ColorJitter: randomly changes brightness, contrast, and saturation
3. Lighting: similar to color jitter but added to the R/G/B channels
4. HorizontalFlip: 50% chance of horizontal flip

The implementations are in:
https://github.com/facebook/fb.resnet.torch/blob/master/datasets/transforms.lua

We do the same normalization, but don't add any noise during validation and in `extract_features.lua` to improve accuracy.

The large snapshots are because the code doesn't call `:clearState()` on the network, which it should do. 

You can create a smaller saved model from one of the checkpoints by:

```lua
require 'cudnn'
m = torch.load('model-best.t7')
m:clearState()
torch.save('small.t7', m)
```

**Comment by afshindn** [comment]
Thanks. This is helpful.
